### PR TITLE
Update messages style according to GUI specification

### PR DESCRIPTION
**Pull request by solth**
This pull request changes the design of the static messages displayed in Kitodo.Production to the current GUI design specification.

Info message example:
Bildschirmfoto 2019-07-01 um 09 50 25

Error message example:
Bildschirmfoto 2019-07-01 um 09 50 45

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 1]
```diff
@@ -731,19 +732,29 @@ public void onDragDrop(TreeDragDropEvent event) {
...
+            }
+        } catch (ClassCastException exception) {
             Helper.setErrorMessage("Unable to move structure element!");
         }
     }
 
-    private void checkLogicalDragDrop(IncludedStructuralElement dragStructure,
-                                      IncludedStructuralElement dropStructure) {
+    private void checkLogicalDragDrop(StructureTreeNode dragNode, StructureTreeNode dropNode)
+            throws ClassCastException {
```
Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java]: No need to define it here - it is runtime exception.

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 2]
```diff
@@ -731,19 +732,29 @@ public void onDragDrop(TreeDragDropEvent event) {
...
+
+        try {
+            if (dragNode.getDataObject() instanceof IncludedStructuralElement
+                    && dropNode.getDataObject() instanceof IncludedStructuralElement) {
+                checkLogicalDragDrop(dragNode, dropNode);
+            } else if (dragNode.getDataObject() instanceof MediaUnit
+                    && dropNode.getDataObject() instanceof MediaUnit) {
+                checkPhysicalDragDrop(dragNode, dropNode);
+            } else {
+                Helper.setErrorMessage("Unable to move element of type " + dragNode.getLabel() + " to element of type "
```
Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java]: Missing translation message.

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 3]
```diff
@@ -756,20 +767,25 @@ private void checkLogicalDragDrop(IncludedStructuralElement dragStructure,
                     preserveLogical();
                     return;
                 } else {
-                    Helper.setErrorMessage("Parents of structure " + dragStructure.getType()
+                    Helper.setErrorMessage("Parents of structure " + dragNode.getLabel()
```
Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java]: Missing translation message.

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 4]
```diff
@@ -756,20 +767,25 @@ private void checkLogicalDragDrop(IncludedStructuralElement dragStructure,
...
                             + " do not contain structure!");
                 }
             } else {
-                Helper.setErrorMessage("No parents of structure " + dragStructure.getType() + " found!");
+                Helper.setErrorMessage("No parents of structure " + dragNode.getLabel() + " found!");
```
Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java]: Missing translation message.

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 5]
```diff
@@ -756,20 +767,25 @@ private void checkLogicalDragDrop(IncludedStructuralElement dragStructure,
...
-                    + "' NOT allowed as child of structure of type '" + dropStructure.getType() + "'! ");
+            Helper.setErrorMessage(Helper.getTranslation("dataEditor.forbiddenChildElement",
+                    Arrays.asList(dragNode.getLabel(), dropNode.getLabel())));
         }
         show();
     }
 
-    private void checkPhysicalDragDrop(MediaUnit dragUnit, MediaUnit dropUnit) {
+    private void checkPhysicalDragDrop(StructureTreeNode dragNode, StructureTreeNode dropNode)
+            throws ClassCastException {
```
Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java]: No need to define it here - it is runtime exception.

**Comment by Beacze** [comment]
Just general question - title of PR says update messages style but it changes methods in StructurePanel. Is it intended?

**Comment by solth** [comment]
> Just general question - title of PR says update messages style but it changes methods in StructurePanel. Is it intended?

In this case, yes, it is intended. The "Label", which contains the translated type of a structural element used in the error messages, was not available in corresponding methods in the backend. Therefore these methods were slightly adjusted.

**Pull request merged**